### PR TITLE
fix panic when using MixedInstancesPolicy

### PR DESCRIPTION
**Pull request by backjo**

### CODE REVIEW

File: controllers/upgrade.go [hunk 1]
```diff
@@ -503,7 +503,7 @@ func (r *RollingUpgradeContext) IsInstanceDrifted(instance *autoscaling.Instance
 			launchTemplateName      = aws.StringValue(scalingGroup.MixedInstancesPolicy.LaunchTemplate.LaunchTemplateSpecification.LaunchTemplateName)
 			instanceTemplateName    = aws.StringValue(instance.LaunchTemplate.LaunchTemplateName)
 			instanceTemplateVersion = aws.StringValue(instance.LaunchTemplate.Version)
-			templateVersion         = aws.StringValue(scalingGroup.LaunchTemplate.Version)
+			templateVersion         = aws.StringValue(scalingGroup.MixedInstancesPolicy.LaunchTemplate.LaunchTemplateSpecification.Version)
```
Comment (eytan-avisror) [inline controllers/upgrade.go:506]: We need to check both `scalingGroup.MixedInstancesPolicy.LaunchTemplate.LaunchTemplateSpecification.Version`
 and `scalingGroup.LaunchTemplate.Version`

Comment (eytan-avisror) [reply controllers/upgrade.go:506]: Otherwise, when a launch template does not have a mixedInstancePolicy you will get a nil pointer for templateVersion

Comment (eytan-avisror) [reply controllers/upgrade.go:506]: I think we need to do a nil check first to understand whether it's a standard launch template or launch template with mixed instance policy, then we know which struct to access to get the name / version

Comment (eytan-avisror) [reply controllers/upgrade.go:506]: Oh sorry, I see we already do it above - this change is correct

**Comment by backjo** [comment]
@eytan-avisror @shreyas-badiger any chance we could get that hotfix with this in?

**Comment by shreyas-badiger** [comment]
@backjo yes. we can do a minor release soon. I have another bug to fix. Will do it right after that.

**Pull request merged**